lib/engine/matchers/all_keys_matcher.rb

```diff
@@ -15,7 +15,7 @@ def initialize
     # @param key [string] key value to be matched
     #
     # @return [boolean] true for all instances
-    def match?(_key, _attributes)
+    def match?(_key, _split_treatment, _attributes)
```

Comment:
This is an idiom in Ruby: if I start param name with underscore it means that it's not used in the method (although required by the interface). So, since these params are not used in the method they require no docs here. Will add docs in `combining_matcher.rb` though